lstests/core/service.py

```diff
@@ -0,0 +1,1526 @@
+                         extra_configs=self.extra_configs,
+                         extra_args=self.extra_args,
+                         **call_kwargs)
+
+    def stop_cluster(self, **call_kwargs):
+        return self.call("stop_cluster", **call_kwargs)
+
+    def clean_cluster(self, **call_kwargs):
+        return self.call("clean_cluster", **call_kwargs)
+
+    def start_node(self, node_id, extra_args=None, **call_kwargs):
+        return self.call("start_node", node_id=node_id, extra_args=extra_args, **call_kwargs)
+
+    def stop_node(self, node_id, kill_sig=15, check=True, dont_raise=False, **call_kwargs):
+        """kill_sig: 15 for soft kill, 9 for hard kill"""
+        cx = self.call("stop_node", node_id=node_id, kill_sig=kill_sig, **call_kwargs)
+        if check:
+            for _ in range(10):
+                if self.is_node_down(node_id=node_id):
+                    return cx
+                time.sleep(1)
+            else:
+                if not dont_raise:
+                    raise BlockchainError(f"Node {node_id} cannot be properly stopped by singal {kill_sig}.")
+        else:
+            return cx
+
+# --------------- simple queries: queries that are made directly via call() -------------------------------------------
+
+    def get_cluster_running_state(self, **call_kwargs):
+        return self.call("get_cluster_running_state", **call_kwargs)
+
+    def get_cluster_info(self, **call_kwargs):
+        return self.call("get_cluster_info", **call_kwargs)
+
+    def get_info(self, node_id, **call_kwargs):
+        return self.call("get_info", node_id=node_id, **call_kwargs)
+
+    def get_block(self, block_num_or_id, node_id=0, **call_kwargs):
+        return self.call("get_block", node_id=node_id, block_num_or_id=block_num_or_id, **call_kwargs)
+
+    def get_account(self, name, node_id=0, **call_kwargs):
+        return call("get_account", node_id=node_id, name=name, **call_kwargs)
+
+    def get_protocol_features(self, node_id=0, **call_kwargs):
+        return call("get_protocol_features", node_id=node_id, **call_kwargs)
+
+    def get_log_data(self, offset, node_id=0, length=10000, filename="stderr_0.txt", **call_kwargs):
+        return self.call("get_log_data", node_id=node_id, offset=offset, length=length, filename=filename, **call_kwargs)
+
+    def verify_transaction(self, transaction_id, verify_key=None, node_id=0, **call_kwargs):
+        try:
+            return self.call("verify_transaction", node_id=node_id, transaction_id=transaction_id,
+                             verify_key=None, dont_flush=True, **call_kwargs).response_dict[verify_key]
+        except KeyError:
+            return False
+
+# --------------- composite queries: queries that are dependent on simple queries -------------------------------------
+
+    def get_node_pid(self, node_id, **call_kwargs):
+        return self.get_cluster_running_state(**call_kwargs).response_dict["result"]["nodes"][node_id][1]["pid"]
```

Comment:
Thanks for the comment and the code suggestion. I have updated the code with minor differences with from your suggestion.

```python
def log_response_error(func):
    @functools.wraps(func)
    def wrapper(self, *args, **kwargs):
        try:
            return func(self, *args, **kwargs)
        except (KeyError, IndexError, TypeError):
            self.error("Response dictionary may have changed format.")
            raise
    return wrapper
```

1. It looks like not only `KeyError` but also `IndexError` and `TypeError` can be triggered by a format change in response dictionary.

2. It seems that `self` should be put as a parameter of `wrapper` instead of `log_response_error`.

3. It seems that in the code suggestion a `return wrapper` is missing.

Thank you.